theano/tensor/elemwise.py

```diff
@@ -1059,35 +1076,110 @@ def _c_all(self, node, nodename, inames, onames, sub):
                                      fortran=alloc_fortran)
             alloc += cgen.make_checks([list(range(nnested))], [odtype],
                                       dict(sub, lv0=oname))
-        olv_index = i  # index of the last output
+        # index of the last output
```

Comment:
line 1075, put in the "alloc" bloc a call to cgen.make_alloc(). I was thinking it was called for all output, but you are right, they are called only for output that we aren't inplace on.

But this cause problems I think. we want to have all the elemwise version independent of the inplace use the same c code. So we should not check which output is inplace or not. So I think we should change the logic to something like:

- set all output that are inplace.
- then check all outputs that have not been set and allocated them if they aren't preallocated.